### PR TITLE
Update to bypass detection as of July 2024

### PR DESCRIPTION
**Pull request by BluestoneDE**
## Notes
Apparently you cannot simulate clicks on the skip button anymore.
I tired and at least for now we can skip the video manually by changing the `video.currentTime`.
This however results in the player showing the download button for specific ads with install links.
grafik
Notice the piechart timer in the top right corner. This ad goes away automatically by Youtube itself after 5 seconds.
Also I neatly ordered the popup inputs into a table.

## Changes
* Added option to toggle ad visibility (hide by default)
* Added option to force skip ad after 2s (off by default)
* Temporarily disabled `SkipButton.click()` because it gets detected

### CODE REVIEW

**Review by MartinBraquet — COMMENTED** [review]
Thanks again for a nice upgrade! One comment I have is that the current version works fine on my mobile and desktop (skip button clicked without detection from youtube), and since the daily number of users didn't fall these past days, I would assume that the current version is still working well for some non-negligible amount of users.
So, I would prefer to keep the default behavior as is done in the current version (accelerate ad, auto skip button on, hide video, mute video, ad playback rate = 12), and have all your nice upgrades available via toggle buttons in the extension settings. In practice, this would mean:
- Ad video hidden or not (default: hidden, already done in your PR I believe)
- One dropdown for skip ad buttons, one among:
  -  clicking on the skip button (default)
  - changing the time to the end of the ad after 2s
  - Do nothing when the skip button shows
  This would in essence replace your skipWanted check with a dropdown with three possibilities. The config value, named for example skipButtonBehavior, would be a string, which we would use in content.js to decide what to do when `(skipButton && !skipButton.clicked)`.

Let me know what you think and if that seems reasonable to fix issue #7 .

**Comment by MartinBraquet** [comment]
I'll come back tomorrow, as I'm on the central EU time.

**Comment by BluestoneDE** [comment]
Yeah, I cannot do 12x speed with the temporary skip click workaround. 5x is good enough that it doesn't get detected and we want this extension to "work out of the box". I even allowed for the video to not get sped up at all via a new minimum for that option.
I'm central European too but I care a lot less about sleep deprivation.

File: content.js [hunk 1]
```diff
@@ -70,48 +70,61 @@ async function checkForAds() {
...
                         skipButton.clicked = true;
+                        // cannot simulate click on skip button, it will be detected
+                        // skipButton.click();
+                        skipVideo();
                         console.log("YtAd detected, clicking skip button (" + skipButton.className + ")");
                     }
                 }
+
+                function skipVideo() {
+                    if (video && video.detected) {
```
Comment (MartinBraquet) [inline content.js:120]: Here we need to be sure that the video is still an ad, as youtube switches from ad to real video without changing much of the fields (not sure .detected will be erased when the ad is finished). It is an issue as you are running this after 2s, so the video can be the real video at that point I think. This triggered issues in the past where users would get their real video skipped. So, one extra safety check would be to make sure the video duration is the same as the initial ad duration.

Comment (BluestoneDE) [reply content.js:120]: I have tested this plenty of times. The video does change and .detected gets erased.

**Comment by BluestoneDE** [comment]
Alright I did a lot. Still no idea how to skip this install links problem.
YouTube also detects a click simulation here because it's the same "Skip Ad" button.

**Comment by BluestoneDE** [comment]
Alright I did a lot. Still no idea how to skip this install link problem.
YouTube also detects a click simulation here because it's the same "Skip Ad" button.

**Review by MartinBraquet — APPROVED** [review]
Looks great, and nice UI for the popup!
I just pushed a commit with minor updates. The only main behavior change is that the boost after 2s is now optional and disabled by default. As the main goal at this point is to avoid youtube detection, I think it's better to do as few "minor" modifs as possible. As long as the ad is accelerated (and possibly skipped via button or go to end time when the skip button appears), the extension should work well out of the box.
Let me know if the PR looks ok to you now.

**Comment by MartinBraquet** [comment]
Also, it would be helpful if you can write down, in this PR or another one, a few bullet points in the readme about which methods don't work anymore as of today. Knowledge of failures is also valuable knowledge, which will help guide future development.

**Comment by BluestoneDE** [comment]
Looks good to me. Basically you undid my changes in 5c3d34d with the default option of it being turned off.
I tested this version with your changes and it works as expected.

**Comment by MartinBraquet** [comment]
Right, boosting to the ad end should probably be done regardless of the ad skip behavior. So what you just pushed makes sense to me. Will wait for an answer an #7 . If the issue doesn't get answered, I'll merge tomorrow.

**Pull request merged**